release.sh

```diff
@@ -3,39 +3,98 @@
 set -e
 set -x
 
-if [ ! -d ${HOME}/google-cloud-sdk ]; then
-  wget https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-sdk-206.0.0-linux-x86_64.tar.gz -O install.tar.gz
-  tar -xvzf install.tar.gz
+KEY_FILE=
+SPIN_GCS_BUCKET_PATH=
+VERSION=
+
+
+function process_args() {
+  while [[ $# > 0 ]]
+  do
+    local key="$1"
+    shift
+    case $key in
+      --gcs_bucket_path)
+        SPIN_GCS_BUCKET_PATH="$1"
+        shift
+        ;;
+      --key_file)
+        KEY_FILE="$1"
+        shift
+        ;;
+      --version)
+        VERSION="$1"
+        shift
+        ;;
+      --help|-help|-h)
+        print_usage
+        exit 13
+        ;;
+      *)
+        echo "ERROR: Unknown argument '$key'"
+        exit -1
+    esac
+  done
+}
+
+
+function print_usage() {
+    cat <<EOF
+usage: $0 [--gcs_bucket_path <path>]
+    [--key_file <key file path>] [--version <version>]
+
+
+    If run with no arguments you will be prompted for cloud provider and region
+
+    --gcs_bucket_path <arg>      Path to the GCS bucket to write the `spin` binaries to.
 
-  ./google-cloud-sdk/install.sh -q
+    --key_file <arg>             Service account JSON key file to use to upload `spin`
+                                       binaries.
+
+    --version <arg>              Version to tag the `spin` binaries with.
+EOF
+}
+
+
+process_args "$@"
+
+# Google cloud sdk installation from
+# https://cloud.google.com/sdk/docs/downloads-interactive.
+if [ ! -d ${HOME}/google-cloud-sdk ]; then
```

Comment:
why a user-installed gcloud and not the machine default if there is one?

Reply:
Where is that gcloud distribution installed?

Reply:
Should this just be a `which gcloud` instead?